lib/membrane_h264_ffmpeg/parser.ex

```diff
@@ -123,16 +124,18 @@ defmodule Membrane.H264.FFmpeg.Parser do
   end
 
   @impl true
-  def handle_process(:input, %Buffer{payload: payload, metadata: metadata}, ctx, state) do
+  def handle_process(:input, buffer, ctx, state) do
     payload =
       if ctx.pads.output.start_of_stream? do
-        payload
+        buffer.payload
       else
-        state.first_frame_prefix <> payload
+        state.first_frame_prefix <> buffer.payload
       end
 
-    with {:ok, sizes, resolution_changes} <- Native.parse(payload, state.parser_ref) do
-      {bufs, state} = parse_access_units(payload, sizes, metadata, state)
+    with {:ok, sizes, output_picture_numbers, resolution_changes} <-
+           Native.parse(payload, state.parser_ref) do
+      metadata = %{buffer_metadata: buffer.metadata, pts: buffer.pts, dts: buffer.dts}
```

Comment:
can't we just keep and pass around the buffer struct?

Reply:
I'm not sure if that is reasonable from the performance perspective because of the `:payload` field.

Reply:
Actually, this is not too readable too - the metadata field in the state is called the same as the metadata field in buffer and it's hard to grasp what it actually contains. The performance shouldn't be affected unless we accumulate lot's of buffers - actually, it may be improved because instead of setting three values, we could set the payload only. Finally, if we happen to add another field to the buffer, this won't work out of the box. Maybe we could name the variable buffer_template, and possibly set the payload to `nil` until it's updated?